src/patternfly/components/CustomSearchInput/custom-search-input.scss

```diff
@@ -0,0 +1,120 @@
+.pf-c-custom-search-input {
+  // Base border
+  --pf-c-custom-search-input__text--before--BorderWidth: var(--pf-global--BorderWidth--sm);
+  --pf-c-custom-search-input__text--before--BorderColor: var(--pf-global--BorderColor--300);
+
+  // Bottom border
+  --pf-c-custom-search-input__text--after--BorderBottomWidth: var(--pf-global--BorderWidth--sm);
+  --pf-c-custom-search-input__text--after--BorderBottomColor: var(--pf-global--BorderColor--200);
+  --pf-c-custom-search-input--hover__text--after--BorderBottomColor: var(--pf-global--primary-color--100);
+  --pf-c-custom-search-input__text--focus-within--after--BorderBottomWidth: var(--pf-global--BorderWidth--md);
+  --pf-c-custom-search-input__text--focus-within--after--BorderBottomColor: var(--pf-global--primary-color--100);
+
+  // Search input
+  --pf-c-custom-search-input__text-input--PaddingTop: var(--pf-global--spacer--form-element);
+  --pf-c-custom-search-input__text-input--PaddingRight: var(--pf-global--spacer--sm);
+  --pf-c-custom-search-input__text-input--PaddingBottom: var(--pf-global--spacer--form-element);
+  --pf-c-custom-search-input__text-input--PaddingLeft: var(--pf-global--spacer--xl);
+  --pf-c-custom-search-input__text-input--MinWidth: 6ch;
+
+  // Icon
+  --pf-c-custom-search-input__icon--Left: var(--pf-global--spacer--sm);
+  --pf-c-custom-search-input__icon--Color: var(--pf-global--Color--200);
+  --pf-c-custom-search-input__text--hover__icon--Color: var(--pf-global--Color--100);
+  --pf-c-custom-search-input__text-input--focus--icon--Color: var(--pf-global--Color--100);
+  --pf-c-custom-search-input__icon--TranslateY: -50%;
+
+  // Utilities
+  --pf-c-custom-search-input__utilities--MarginRight: var(--pf-global--spacer--sm);
+  --pf-c-custom-search-input__utilities--child--MarginLeft: var(--pf-global--spacer--xs);
+  --pf-c-custom-search-input__utilities--c-button--PaddingRight: var(--pf-global--spacer--xs);
+  --pf-c-custom-search-input__utilities--c-button--PaddingLeft: var(--pf-global--spacer--xs);
+
+  position: relative;
+  display: flex;
+  padding: var(--pf-c-custom-search-input--PaddingTop) var(--pf-c-custom-search-input--PaddingRight) var(--pf-c-custom-search-input--PaddingBottom) var(--pf-c-custom-search-input--PaddingLeft);
+
+  &:hover {
+    --pf-c-custom-search-input__text--after--BorderBottomColor: var(--pf-c-custom-search-input--hover__text--after--BorderBottomColor);
+  }
+}
+
+.pf-c-custom-search-input__text {
+  flex: 1;
+
+  &::before,
+  &::after {
```

Comment:
could you not draw the border with just `::after` like other components ?

Reply:
If it's to handle both the base (grey) border and hover/focus (blue) border, it's because using a single border creates an angle where a 1px and 2px border combine. We try to avoid that, so we use a separate border when borders of different colors combine like that.

<img width="105" alt="Screen Shot 2020-07-09 at 9 59 47 AM" src="https://user-images.githubusercontent.com/35148959/87056255-fc612180-c1ca-11ea-8ed4-af271c883476.png">

Reply:
Didn't realize! thanks for explaining